pkg/expr/sql/parser_allow.go

```diff
@@ -63,6 +63,9 @@ func allowedNode(node sqlparser.SQLNode) (b bool) {
 	case sqlparser.BoolVal:
 		return
 
+	case *sqlparser.CaseExpr, *sqlparser.When:
+		return
+
```

Comment:
I added these at the same line numbers as you have them in #101820 - so you shouldn't see a merge-conflict here.